tests/vmi_networking_test.go

```diff
@@ -708,26 +708,28 @@ var _ = Describe("[rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				clientVMI = masqueradeVMI([]v1.Port{}, ipv4NetworkCIDR)
 				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(clientVMI)
 				Expect(err).ToNot(HaveOccurred())
-				tests.WaitUntilVMIReady(clientVMI, tests.LoggedInCirrosExpecter)
+				clientVMI = tests.WaitUntilVMIReady(clientVMI, tests.LoggedInCirrosExpecter)
+				Expect(clientVMI.Status.Interfaces).To(HaveLen(1))
+				Expect(clientVMI.Status.Interfaces[0].IPs).NotTo(BeEmpty())
 			}
 
 			serverVMI = masqueradeVMI(ports, ipv4NetworkCIDR)
 			serverVMI.Labels = map[string]string{"expose": "server"}
 			_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(serverVMI)
 			Expect(err).ToNot(HaveOccurred())
-
-			tests.WaitUntilVMIReady(serverVMI, tests.LoggedInCirrosExpecter)
-
-			serverVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(serverVMI.Name, &v13.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(len(serverVMI.Status.Interfaces)).To(Equal(1))
+			serverVMI = tests.WaitUntilVMIReady(serverVMI, tests.LoggedInCirrosExpecter)
+			Expect(serverVMI.Status.Interfaces).To(HaveLen(1))
 			Expect(serverVMI.Status.Interfaces[0].IPs).NotTo(BeEmpty())
 
 			By("starting a tcp server")
-			tcpPort := "8080"
-			err = tests.CheckForTextExpecter(serverVMI, createExpectStartTcpServer(tcpPort), 30)
+			tcpPort := 8080
+			tests.StartTCPServer(serverVMI, tcpPort)
 			Expect(err).ToNot(HaveOccurred())
 
+			for _, clientIP := range clientVMI.Status.Interfaces[0].IPs {
+				Expect(tests.PingFromVMConsole(serverVMI, clientIP)).To(Succeed())
+			}
```

Comment:
Why do you need this direction as well? What is this testing that we have not tested already?

Reply:
The only reason I see is that the clientVMI is configured with masquerade binding but without ports (`clientVMI = masqueradeVMI([]v1.Port{}, ipv4NetworkCIDR)` so we check that it's not closing ICMP, but we can drop this since it's not done a bridge binding counterpart test.

Reply:
In case the current test configuration might have any problem which this check would detect, i would suggest not to remove it.
Its fast check and this is one of the main tests of connectivity.

Reply:
@qinqon these are table tests, there is one that uses specific ports and all the rest without, so this is checked as well with pinging the server (please let me know if I understood you incorrectly here).

@oshoval , please point to a specific scenario which is not checked here. You need to reason why the assertions are needed here, arbitrary checks may add a level of protection but makes the whole test unfocused and a mess.
The test uses a client to check connectivity to the server. The client is static, the server changes configurations so each is checked... please show the missing scenario (like qinqon did above) so we can reason for this.

Reply:
> @qinqon these are table tests, there is one that uses specific ports and all the rest without, so this is checked as well with pinging the server (please let me know if I understood you incorrectly here).
> 
> @oshoval , please point to a specific scenario which is not checked here. You need to reason why the assertions are needed here, arbitrary checks may add a level of protection but makes the whole test unfocused and a mess.
> The test uses a client to check connectivity to the server. The client is static, the server changes configurations so each is checked... please show the missing scenario (like qinqon did above) so we can reason for this.

I dont have a specific reason in mind, but i do think its worth to keep it, to check both symmetric traffic, with both protocols.
this test is an important test of masquerade.
I dont think it adds more than 2 seconds, or a mass in this specific case.

Reply:
Removed, since we don't know what defects we can catch with this.

Reply:
I think the fact i dont have a specific scenario doesnt mean we should remove it
its a different protocol, we shouldn't limit the tests so strictly imo.

Reply:
Based on this algorithm and reasoning we can add tests for ftp, sip, udp, traceroute and many more. All use special protocols.
BTW, it was never there, so this is not about removing but about not adding.

Reply:
i see it otherwise, we added ICMP , i think it worth adding bidirectional

Reply:
Last time I checked, an ICMP echo was a bidirectional check.

Reply:
i think one can disable ICMP initiating and just enable responses. (i think i saw this configuration at the past)
again i dont think it worth not checking it just because a round trip saying it works.

Reply:
I’m sorry but you have not managed to provide any specific scenarios for meriting adding the check. A lot may go wrong, but we still limit ourself to the reasonable checks.
The test is already overloaded with checks, making it hard to maintain and read.

The ping addition aimed to test the icmp protocol passing through the masquerade configuration, with and without ports defined, which is a reasonable reason to have.
We should stop there.

/hold cancel

Reply:
https://superuser.com/questions/1111392/ping-only-working-in-one-direction
or just google "can ping work just from one direction" and you see more results that you might accept more,
i saw this phenomena in the past (previous companies).

One can even say, we already check tcp server from client and its also round trip, so why use ICMP (unless because its a different protocol, which in this case i think we should check it both directional)

Reply:
You keep on ref general possibilities but not a specific case that ma happen here.
We do not check tcp in both directions, we do not check udp because it behaves like tcp in this case, we surely should not check unproven theories.